src/app/components/map/map.component.ts

```diff
@@ -498,16 +498,16 @@ export class MapComponent implements OnInit, OnDestroy {
   }
 
   public async deleteDevice(): Promise<void> {
-    this.modalService.confirm(this.confirmTitleString, this.confirmBodyString)
-      .then(async confirmed => {
-        if (confirmed) {
-          try {
-            await this.deviceService.unregister(this.selectedDevice._id);
-          } catch (e) {
-            this.alertService.error(e.error.message);
-          }
+    try {
+      const confirmed = await this.modalService.confirm(this.confirmTitleString, this.confirmBodyString);
```

Comment:
Valt me op dat we op een aantal plekken in de code een geneste try-catch structuur hebben, speciaal voor het dismissen van de modal. In denk dat dit netter kan door het catchen in de `modalService` zelf te doen. Door `modalService.confirm` niet langer te laten `throw`en kan de code op veel plekken (zoals deze) opgeschoond worden.

De methode in de `modalService` zou er dan ongeveer zo kunnen uitzien:
```
  public async confirm(title: string, message: string, ...): Promise<boolean> {
    const modalRef = this.modalService.open(...);
    ...

    let result = false;
    try {
      result = await modalRef.result;
    } catch { }

    return result;
  }
  ```